instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py

```diff
@@ -158,6 +166,33 @@ def response_hook(span, request_obj, response):
 _ResponseHookT = Optional[Callable[[Span, PreparedRequest, Response], None]]
 
 
+def _detect_synthetic_user_agent(user_agent: str) -> Optional[str]:
```

Comment:
I think this and the patterns can go into `opentelemetry-util-http` since there's nothing requests specific in this code and I guess would be nice to keep the behaviour consistent